MODULE_NOT_FOUND during `docker cp` command

Command I'm running
`act -P ubuntu-latest=nektos/act-environments-ubuntu:18.04 `

Error output
```
 ☁  git clone 'https://github.com/gittools/actions' # ref=v0.9.2
[]   🐳  docker cp src=/home/user/.cache/act/gittools-actions-gitversion-setup@v0.9.2 dst=/actions/
| internal/modules/cjs/loader.js:985
|   throw err;
|   ^
| 
| Error: Cannot find module '/actions/gittools-actions-gitversion-setup@v0.9.2/dist/github/gitversion/setup/bundle.js'
|     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:982:15)
|     at Function.Module._load (internal/modules/cjs/loader.js:864:27)
|     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
|     at internal/main/run_main_module.js:18:47 {
|   code: 'MODULE_NOT_FOUND',

```
it looks like the code from gittools-actions-gitversion-setup@v0.9.2 is copied to /actions/ in the container but the action javascript expects the original directory name to be in the path. So the cp should be `cp /home/user/.cache/act/gittools-actions-gitversion-setup@v0.9.2 /actions/gittools-actions-gitversion-setup@v0.9.2`

--- a/pkg/runner/runner_test.go
+++ b/pkg/runner/runner_test.go
@@ -53,6 +53,7 @@ func TestRunEvent(t *testing.T) {
 		{"matrix", "push", ""},
 		{"commands", "push", ""},
 		{"workdir", "push", ""},
+		{"issue-228", "push", ""}, // TODO [igni]: Remove this once everything passes
 	}
 	log.SetLevel(log.DebugLevel)
 

--- a/pkg/runner/step_context.go
+++ b/pkg/runner/step_context.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"path"
 	"path/filepath"
 	"regexp"
 	"runtime"
@@ -266,7 +267,11 @@ func (sc *StepContext) runAction(actionDir string, actionPath string) common.Exe
 		switch action.Runs.Using {
 		case model.ActionRunsUsingNode12:
 			if step.Type() == model.StepTypeUsesActionRemote {
-				err := rc.JobContainer.CopyDir(containerActionDir+string(filepath.Separator), actionDir)(ctx)
+				err := removeGitIgnore(actionDir)
+				if err != nil {
+					return err
+				}
+				err = rc.JobContainer.CopyDir(containerActionDir+string(filepath.Separator), actionDir)(ctx)
 				if err != nil {
 					return err
 				}
@@ -350,4 +355,21 @@ func newRemoteAction(action string) *remoteAction {
 		ra.Ref = matches[6]
 	}
 	return ra
+}
+
+// https://github.com/nektos/act/issues/228#issuecomment-629709055
+// files in .gitignore are not copied in a Docker container
+// this causes issues with actions that ignore other important resources
+// such as `node_modules` for example
+func removeGitIgnore(directory string) error {
+	gitIgnorePath := path.Join(directory, ".gitignore")
+	if _, err := os.Stat(gitIgnorePath); err == nil {
+		// .gitignore exists
+		log.Debugf("Removing %s before docker cp", gitIgnorePath)
+		err := os.Remove(gitIgnorePath)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }

--- a/pkg/runner/testdata/issue-228/main.yaml
+++ b/pkg/runner/testdata/issue-228/main.yaml
@@ -7,7 +7,7 @@ jobs:
   kind:
     runs-on: ubuntu-latest
     steps:
-      - run: apt install git -y # setup git credentials will fail otherwise
+      - run: apt-get update -y && apt-get install git -y # setup git credentials will fail otherwise
       - name: Setup git credentials
         uses: fusion-engineering/setup-git-credentials@v2
         with:

--- a/pkg/runner/testdata/issue-228/main.yaml
+++ b/pkg/runner/testdata/issue-228/main.yaml
@@ -1,0 +1,14 @@
+name: issue-228
+
+on:
+  - push
+
+jobs:
+  kind:
+    runs-on: ubuntu-latest
+    steps:
+      - run: apt install git -y # setup git credentials will fail otherwise
+      - name: Setup git credentials
+        uses: fusion-engineering/setup-git-credentials@v2
+        with:
+          credentials: https://test@github.com/

